qiskit/synthesis/evolution/qdrift.py

```diff
@@ -0,0 +1,99 @@
+         `arXiv:quant-ph/1811.08017 <https://arxiv.org/abs/1811.08017>`_
+    """
+
+    def __init__(
+        self,
+        reps: int = 1,
+        insert_barriers: bool = False,
+        cx_structure: str = "chain",
+        atomic_evolution: Optional[
+            Callable[[Union[Pauli, SparsePauliOp], float], QuantumCircuit]
+        ] = None,
+    ) -> None:
+        r"""
+        Args:
+            reps: The number of times to repeat the Trotterization circuit.
+            insert_barriers: Whether to insert barriers between the atomic evolutions.
+            cx_structure: How to arrange the CX gates for the Pauli evolutions, can be
+                "chain", where next neighbor connections are used, or "fountain", where all
+                qubits are connected to one.
+            atomic_evolution: A function to construct the circuit for the evolution of single
+                Pauli string. Per default, a single Pauli evolution is decomopsed in a CX chain
+                and a single qubit Z rotation.
+        """
+        super().__init__(1, reps, insert_barriers, cx_structure, atomic_evolution)
+        self.sampled_ops = None
+
+    def synthesize(self, evolution):
+        # get operators and time to evolve
+        operators = evolution.operator
+        time = evolution.time
+
+        if not isinstance(operators, list):
+            pauli_list = [(Pauli(op), coeff) for op, coeff in operators.to_list()]
+            coeffs = [np.real(coeff) for op, coeff in operators.to_list()]
+        else:
+            pauli_list = [(op, 1) for op in operators]
+            coeffs = [1 for op in operators]
+
+        # We artificially make the weights positive
+        weights = np.abs(coeffs)
+        lambd = np.sum(weights)
+
+        num_gates = 2 * (lambd ** 2) * (time ** 2) * self.reps
+        # The protocol calls for the removal of the individual coefficients,
+        # and multiplication by a constant evolution time.
+        evolution_time = lambd * time / num_gates
+        self.sampled_ops = algorithm_globals.random.choice(
+            np.array(pauli_list, dtype=object),
+            size=(int(np.ceil(num_gates)),),
+            p=weights / lambd,
+        )
+
+        # pylint: disable=cyclic-import
+        from qiskit.circuit.library.pauli_evolution import PauliEvolutionGate
+        from qiskit.opflow import PauliOp
+
+        # Build the evolution circuit using the LieTrotter synthesis with the sampled operators
+        lie_trotter = LieTrotter(
+            insert_barriers=self.insert_barriers, atomic_evolution=self.atomic_evolution
+        )
+        # We pass time=1 because the time is already contained in the coefficients of the Paulis.
+        evolution_circuit = PauliEvolutionGate(
+            sum(PauliOp(op) for op, coeff in self.sampled_ops),
+            time=evolution_time,
+            synthesis=lie_trotter,
+        ).definition
```

Comment:
This comment might be out-of-date now?  It doesn't seem to pass `time=1` anywhere.

Reply:
oops I've removed it now - thanks for noticing.